includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -241,34 +242,46 @@ public function process_apple_pay_verification() {
 	 * @since 4.0.6
 	 */
 	public function admin_notices() {
-		if ( ! $this->stripe_enabled ) {
+		if ( ! $this->stripe_enabled || ! $this->payment_request ) {
 			return;
 		}
 
 		if ( ! current_user_can( 'manage_woocommerce' ) ) {
 			return;
 		}
 
-		if ( $this->payment_request && ! empty( $this->apple_pay_verify_notice ) ) {
-			$allowed_html = array(
-				'a' => array(
-					'href'  => array(),
-					'title' => array(),
-				),
-			);
-
-			echo '<div class="error stripe-apple-pay-message"><p>' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html ) . '</p></div>';
+		$empty_notice = empty( $this->apple_pay_verify_notice );
+		if ( $empty_notice && ( $this->apple_pay_domain_set || empty( $this->secret_key ) ) ) {
+			return;
 		}
 
 		/**
 		 * Apple pay is enabled by default and domain verification initializes
 		 * when setting screen is displayed. So if domain verification is not set,
 		 * something went wrong so lets notify user.
 		 */
-		if ( ! empty( $this->secret_key ) && $this->payment_request && ! $this->apple_pay_domain_set ) {
+		$allowed_html                      = array(
+			'a' => array(
+				'href'  => array(),
+				'title' => array(),
+			),
+		);
+		$error_from_stripe                 = $empty_notice ? '' : '<p>' . wp_kses( make_clickable( esc_html( $this->apple_pay_verify_notice ) ), $allowed_html ) . '.</p>';
```

Comment:
I don't think we need a `.` at the end here, since it looks like some messages already have one (see [screenshot](https://cloudup.com/cXBarovKk2D)) and we're already isolating it from our own strings.

Reply:
Fixed in dff28b197184c8bc2803dfe3314751f9b6666075 🎉